src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -2217,5 +2217,35 @@ public void M(string s)
             Assert.Equal("s", symbol.Name);
             Assert.Equal(SpecialType.System_String, symbol.Type.SpecialType);
         }
+
+        [Fact, WorkItem(37879, "https://github.com/dotnet/roslyn/issues/37879")]
+        public void MissingSymbols_ReinferedParent()
+        {
+            var source = @"
+class C
+{
+    public void A<T>(T t) where T:class
+    {
+        var c = new F<T>[] { }.Select(v => new { Value = v.Item }).ToArray();
+    }
+    private class F<T>
+    {
+        public F(T oldItem) => Item = oldItem;
+        public T Item { get; }
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var select = root.DescendantNodes().OfType<IdentifierNameSyntax>().Where(i => i.Identifier.ValueText == "Select").Single();
+            var symbolInfo = model.GetSymbolInfo(select);
+
+            Assert.Null(symbolInfo.Symbol);
+            Assert.Empty(symbolInfo.CandidateSymbols);
```

Comment:
>symbolInfo.CandidateSymbols [](start = 25, length = 27)

I think it would be good to have a test where we can observe that a non-empty set is preserved, including a success case (do we have one already?).

Reply:
I'll try to find a repro, but the nature of this bug means, I think, that this would be difficult. The issue was that because there was a single "method" symbol, and semantic info wasn't being correctly preserved, the semantic model thought overload resolution succeeded. If there are multiple candidates, then the semantic model could not have been confused.

---
In reply to: [314007446](https://github.com/dotnet/roslyn/pull/37964#discussion_r314007446) [](ancestors = 314007446)